### ISSUE TITLE
Reject viruses (RSpamD + ClamAV)

### ISSUE THREAD

**Issue by chibenwa** [linagora/james-project#4596]
In rspamd extensions, start clamAV in a docker, start rspamD, configured to use SpamAV as an anti-virus

RSpamD scanner mailet MUST reject emails containing viruses:

```
<mailet matcher="All" class="RSpamDScanner">
    <spamProcessor>spam</spamProcessor>
</mailet>
```

If `spamProcessor` option is specified, then emails containing viruses will be redirected to this processor. By default, if unspecified no specific handling is done for viruses.

Modify `AnalysisResult` to carry over analysis results from ClamAV.

**Comment by chibenwa** [comment]
As a user I should NEVER receive an Email containing a Virus
(as this could be a major security issue)

What to do with email with viruses?

 - store them in a mail repository and not deliver them
 - Remove attachment, modify the subject with [VIRUS] and add headers...

This could be a configurable behaviour specified in mailetcontainer.xml `spam` processor, so via configuration

**Comment by quantranhong1999** [comment]
TLDR: RSpamD does not explicitly return the virus scanning result.

It looks like RSpamD checkv2 does not explicitly return the virus checking result. No other endpoint returns this also.

By default, RSpamD consider virus scanning result as criteria with configurable score:
![image.png](https://images.zenhubusercontent.com/5f7fe3b8b67a756e7c63d181/5ec1f4dd-c868-4210-89cb-bc0a64dd9f27)

Beside that, RSpamD support enforce action (eg reject) when a message have virus (cf: https://rspamd.com/doc/modules/antivirus.html#configuration) with a configurable message (e.g `message = '${SCANNER}: virus found: "${VIRUS}"';`). 
However, I am not sure if RSpamD returns this message in `/checkv2` response so we can parse it.
Otherwise, we can check if `checkv2` response contains `CLAM_VIRUS` symbols to know if it is a virus mail or not. I did not succeed in resulting a virus mail to check this though.

Will check more...

**Comment by chibenwa** [comment]
I would expect RSpamD to add symbols for anti-virus checks in the `/checkv2` endpoint. So this requires a bit of response parsing.

Can you publish and share your working branch here?

**Comment by quantranhong1999** [comment]
> Can you publish and share your working branch here?

https://github.com/quantranhong1999/james-project/commits/rspamd-clamav-integration
Not much yet. Likely will need to parse the virus symbol then.

**Comment by chibenwa** [comment]
> Likely will need to parse the virus symbol then.

Yes that is what I would expect.

**Comment by quantranhong1999** [comment]
https://github.com/apache/james-project/pull/1126

**Comment by chibenwa** [comment]
(TOLD YOU :-p )
